presto-router/src/main/java/com/facebook/presto/router/cluster/ClusterStatusTracker.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.cluster;
+
+import com.facebook.airlift.log.Logger;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+
+import javax.annotation.PostConstruct;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.airlift.concurrent.Threads.threadsNamed;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.Sets.difference;
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+
+public class ClusterStatusTracker
+{
+    private static final Logger log = Logger.get(ClusterStatusTracker.class);
+
+    private final ClusterManager clusterManager;
+    private final RemoteInfoFactory remoteInfoFactory;
+    private final ScheduledExecutorService queryInfoUpdateExecutor;
+
+    // Cluster status
+    private final ConcurrentHashMap<URI, RemoteClusterInfo> remoteClusterInfos = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<URI, RemoteQueryInfo> remoteQueryInfos = new ConcurrentHashMap<>();
+
+    @Inject
+    public ClusterStatusTracker(
+            ClusterManager clusterManager,
+            RemoteInfoFactory remoteInfoFactory)
+    {
+        this.clusterManager = requireNonNull(clusterManager, "clusterManager is null");
+        this.remoteInfoFactory = requireNonNull(remoteInfoFactory, "remoteInfoFactory is null");
+        this.queryInfoUpdateExecutor = newSingleThreadScheduledExecutor(threadsNamed("query-info-poller-%s"));
+    }
+
+    @PostConstruct
+    public void startPollingQueryInfo()
+    {
+        clusterManager.getAllClusters().stream()
```

Comment:
nits: `stream()` is probably not needed

Reply:
Yep. I removed the unnecessary `stream()` from the code (there's another place I used it redundantly and I also removed it from there).